### PR TITLE
Build remediation roles

### PR DESCRIPTION
**Pull request by mpreisler**
This PR introduces remediation roles. The idea comes from Ansible galaxy roles. These are files that will put your machine into some state. We have all the necessary data to generate idempotent bash and ansible remediation roles. I figured it would be a good idea to start doing that, as a first step before we start continuous submissions to the Ansible galaxy repository.

Let me know what you think. Once we agree how this will work I will add more profiles from other products. We could even build a role for every single profile for both bash and ansible but that might be excessive.

NOTE: I am building bash and ansible roles. IMO it makes no sense to build anaconda roles. The anaconda language is OSCAP Anaconda Addon specific and there is no way to run it outside of OAA.

### CODE REVIEW

**Comment by jan-cerny** [comment]
@mpreisler  Great!

I would build role for each profile and I would adjust the Cmake so that people won't have to add another lines to Cmakelists when they add new profile and want to build remediations.
I don't think it's excessive. We already build HTML guides for all profile and nobody protests. 
However we might want to fix https://github.com/OpenSCAP/openscap/issues/720 in OpenSCAP.

**Comment by shawndwells** [comment]
+1 to @jan-cerny idea on having remediations built as part of the core build. No need to duplicate profile names everywhere.... gets confusing and annoying (did I build tables for STIG? what about remediations? what about guides?).

**Comment by mpreisler** [comment]
OK, thanks for the feedback guys!

I will create a script similar to build-all-guides.py that will build remediation roles for every profile in every product.

**Comment by mpreisler** [comment]
Found an issue with building all the roles:

```
Profile "xccdf_org.ssgproject.content_profile_pci-dss_centric" was not found. Get available profiles using:
$ oscap info "/home/mpreisle/d/scap-security-guide/build/ssg-rhel7-ds.xml"
Fatal error encountered when generating role '/tmp/ssg-rhel7-PCIDSS-RHEL-7-role-pci-dss_centric.sh'. Error details:
Command '['oscap', 'xccdf', 'generate', 'fix', '--benchmark-id', 'xccdf_org.ssgproject.content_benchmark_PCIDSS-RHEL-7', '--profile', 'xccdf_org.ssgproject.content_profile_pci-dss_centric', '--template', 'asdf', '/home/mpreisle/d/scap-security-guide/build/ssg-rhel7-ds.xml']' returned non-zero exit status 1
```

Reported https://github.com/OpenSCAP/openscap/issues/722

What do you think, should we build just the first benchmark for now? I guess that's a good compromise.

**Comment by mpreisler** [comment]
It's building and installing. I still want to do refactoring because a lot of build-all-guides and biuld-all-remediation-roles is the same.

**Comment by mpreisler** [comment]
Filed another issue for OpenSCAP about the warnings: https://github.com/OpenSCAP/openscap/issues/723

**Comment by mpreisler** [comment]
@openscap-jenkins test this please

File: cmake/SSGCommon.cmake [hunk 1]
```diff
@@ -733,6 +758,22 @@ macro(ssg_build_product PRODUCT)
...
+       )"
+       COMPONENT doc
+    )
+    install(
+       CODE "
+       file(GLOB ROLE_FILES \"${CMAKE_BINARY_DIR}/roles/ssg-${PRODUCT}-role-*.sh\") \n
+           file(INSTALL DESTINATION \"\${CMAKE_INSTALL_PREFIX}/${SSG_ROLE_INSTALL_DIR}\"
+               TYPE FILE FILES \${ROLE_FILES}
+       )"
+       COMPONENT doc
```
Comment (yuumasato) [inline cmake/SSGCommon.cmake]: I see these roles more as part of the main component, not part of doc.

Comment (mpreisler) [inline cmake/SSGCommon.cmake]: Fixed

File: cmake/SSGCommon.cmake [hunk 2]
```diff
@@ -817,12 +877,28 @@ macro(ssg_build_derivative_product ORIGINAL SHORTNAME DERIVATIVE)
...
+    )
+    install(
+       CODE "
+       file(GLOB ROLE_FILES \"${CMAKE_BINARY_DIR}/roles/ssg-${DERIVATIVE}-role-*.sh\") \n
+           file(INSTALL DESTINATION \"\${CMAKE_INSTALL_PREFIX}/${SSG_ROLE_INSTALL_DIR}\"
+               TYPE FILE FILES \${ROLE_FILES}
+       )"
+       COMPONENT doc
```
Comment (yuumasato) [inline cmake/SSGCommon.cmake]: Crap, didn't notice these, same code in `ssg_build_derivative_product` as in ssg_`build_product`.

Comment (mpreisler) [inline cmake/SSGCommon.cmake]: Crap, also fixed.

**Comment by yuumasato** [comment]
This is great, thank you!
Ack.

**Comment by mpreisler** [comment]
The warnings about remote OVALs were fixed in https://github.com/OpenSCAP/openscap/pull/764

**Pull request merged**